font-codegen/src/fields.rs

```diff
@@ -1132,7 +1132,7 @@ impl Field {
                         // Prevent identity-op clippy error with `1 * size`
                         size_expr
                     }
-                    _ => quote!(  #count_expr * #size_expr ),
+                    _ => quote!( (#count_expr).saturating_mul(#size_expr) ),
```

Comment:
so the one alternative here that I think we should at least discuss is `checked_mul`, where we will just error immediately on overflow.

The possible issue with `saturating_mul` is that we can end up with an array where the last item is truncated; in this case we will end up failing later anyway (see here) https://github.com/googlefonts/fontations/blob/4e5fde8a3da573467a29ca3d46ca00fddc317e1c/read-fonts/src/font_data.rs#L154

but maybe it makes sense to just fail immediately?

Reply:
Trying this, I found that in many situations doing
`(#count_expr).checked_mul(#size_expr).ok_or(ReadError::OutOfBounds)?` compiles ok, because the embedding function has a signature returning `Result<..., ReadError>` but when `computed_len_expr()` is embedded in `record::ComputeSize::compute_size` this one cannot return a `ReadError` and would need some other type of wrapping. 

I am afraid I am not familiar enough with codegen to upgrade embedding sites of `computed_len_expr()` to consistently handle `OutOfBounds` errors. So perhaps we can file that as an issue as potential future improvement and perhaps you or Chad could consider upgrading codegen sites to handle that better?

Reply:
Good catch. I haven’t dug into it but it doesn’t seem unreasonable to change `compute_size()` to return a `Result`. It would likely require fixing up a few manual impls.

Reply:
I've addressed this in #966.